Tutorials/WinForms/Classic/create-persistent-classes-and-connect-xpo-to-database.md

```diff
@@ -0,0 +1,99 @@
+    ```csharp
+    [PersistentAlias("concat(FirstName, ' ', LastName)")]
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Use the same approach to add the `Order` class with the `ProductName`(String), `OrderDate`(DateTime), and `Freight`(decimal) properties. Check our example code in case of any difficulty:  
+  [Order.cs](/Tutorials/WinForms/Classic/CS/DataAccess/Order.cs)  
+  [Order.vb](/Tutorials/WinForms/Classic/VB/DataAccess/Order.vb)
+* Add the `Orders` and `Customer` properties to create a [relationship](https://docs.devexpress.com/XPO/2041/concepts/relationships-between-objects) between the **Customer** and **Order** classes.
+    ```csharp
+    // Customer.cs
+    [Association("CustomerOrders")]
+    public XPCollection<Order> Orders {
+        get { return GetCollection<Order>(nameof(Orders)); }
+    }
+    
+    // Order.cs
+    private Customer fCustomer;
+    [Association("CustomerOrders")]
+    public Customer Customer {
+        get { return fCustomer; }
+        set { SetPropertyValue(nameof(Customer), ref fCustomer, value); }
+    }
+    ```
+* Add a reference to the **System.Configuration** assembly. This assembly contains classes required to read the connection string settings from the application configuration file [Connection Strings and Configuration Files](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings-and-configuration-files).
+* Add the `ConnectionHelper` class and copy the code from the [ConnectionHelper.cs](/Tutorials/WinForms/Classic/CS/DataAccess/ConnectionHelper.cs) ([ConnectionHelper.vb](/Tutorials/WinForms/Classic/VB/DataAccess/ConnectionHelper.vb)) file. In most cases, this generic helper can be used without modifications. The complete documentation about possible connection settings is provided in the XPO documentation: [Connecting to a Data Store](https://docs.devexpress.com/XPO/2020/feature-center/connecting-to-a-data-store).
+* Open the *Program.cs* file and add the `ConnectionHelper.Connect` method call to the `Main` method (the *Program.vb* file does not exist in a VB.NET project by defult, use the `Form1` class constructor, instead):
+    ```csharp
+    [STAThread]
+    static void Main() {
+        Application.EnableVisualStyles();
+        Application.SetCompatibleTextRenderingDefault(false);
+        ConnectionHelper.Connect(false);
+        Application.Run(new MainForm());
+    }
+    ```
+* Open the application configuration file (*app.config*) and add the connection string. The `InMemoryDataStore` provider is used in this tutorial for the sake of simplicity. The complete list of supported XPO providers is provided at [K18445 - How to create a correct connection string for XPO providers](https://www.devexpress.com/Support/Center/Question/Details/K18445).
```

Comment:
"for the sake of simplicity"?

Reply:
" The complete list of supported XPO providers is provided at [K18445 - How to create a correct connection string for XPO providers](https://www.devexpress.com/Support/Center/Question/Details/K18445)." -> Refer to [K18445 - How to create a correct connection string for XPO providers](https://www.devexpress.com/Support/Center/Question/Details/K18445) for a complete list of supported XPO providers.

Reply:
I refined these texts

e05bb553eec9d3f47ca2e049c9175e8dccefcdb6
191b6f0645c8f7ee66098afcca0f76d7d88237b5